README.md

```diff
@@ -1,63 +1,163 @@
+
+The npl-integrations project can be built and run locally using docker-compose.
+
+### Pre-requisites
+
+For OS X and linux systems, make sure the `/etc/hosts` file includes the line `127.0.0.1 keycloak`
+
+![img.png](docs/img.png)
+
+If the line is not present, follow the steps below to add it:
+
+1. Use the `sudo vim /etc/hosts` command to open the file in a text editor. Enter your password when prompted for this admin command.
+2. Navigate to the end of the file, press `i` to enter insert mode and add the line `127.0.0.1 keycloak`.
+3. Save by pressing `Esc`, then type `:wq` and press `Enter`.
+
+### Build & run
+
+To build and run the project, follow the steps below:
+
+1. Run `mvn clean install` to build and generate NPL-API and clients.
+2. Run `docker compose up --build -d` to ensure the python listener service, the python Streamlit UI and the typescript React frontend are build in addition to engine and engine dependencies, then create and start containers.
+
+### Service endpoints
+
+Once the project is running, services run behind the following URLs:
+
+| Service                   | URL                                |
+|---------------------------|------------------------------------|
+| Engine APIs               | http://localhost:12000/            |
+| Swagger UI of Engine APIs | http://localhost:12000/swagger-ui/ |
+| Keycloak admin console    | http://localhost:11000             |
+| Webapp                    | http://localhost:8090              |
+| Inspector                 | http://localhost:8070              |
+
+## Running NPL on Noumena Cloud & local services
+
+The second deployment possibility consists of running the NPL code on Noumena Cloud and local services.
+Noumena offers a cloud-based environment for running NPL code, which can be accessed at [portal.noumena.cloud](https://portal.noumena.cloud).
+
+This setup includes running a python listener, python Streamlit UI & typescript React webapp locally, and Noumena Engine on Noumena Cloud.
+
+### NPL and keycloak
+
+In this setup, NPL code runs on Noumena Cloud. Supporting services of the NPL Engine are deployed alongside the NPL Engine.
+Supporting services include Keycloak for authentication and authorization and databases.
+
+#### Option 1: Using the NPL CLI & terraform
+
+1. Install the NPL CLI by running `make download-cli` in the root directory.
+2. Add the following variables to your shell to configure the NPL CLI and restart your terminal
+
+    ```
+    export NC_BASE_URL=https://portal.noumena.cloud
+    export NC_EMAIL=your.email@your.domain
+    export NC_ENV=PROD
+    ```
+
+3. Run `make create-app` to create the application with name defined in Makefile.
+4. Run `make iam` to provision keycloak on the created application with terraform.
+5. Run `make clear-deploy` to clear pre-existing packages in the app and upload current NPL and migration sources.
+
+#### Option 2: Using the maven plugin
```

Comment:
...NPL-dev for IntelliJ, right? Though this is a different option: https://documentation.noumenadigital.com/tools/build-tools/Engine-Maven-Plugin/#deploy-source

Reply:
https://noumenadigital.atlassian.net/browse/DVX-80